naming/dns_resolver.go

```diff
@@ -192,28 +192,25 @@ type AddrMetadataGRPCLB struct {
 
 // compileUpdate compares the old resolved addresses and newly resolved addresses,
 // and generates an update list
-func (w *dnsWatcher) compileUpdate(newAddrs []*Update) []*Update {
-	update := make(map[Update]bool)
-	for _, u := range newAddrs {
-		update[*u] = true
-	}
-	for _, u := range w.curAddrs {
-		if _, ok := update[*u]; ok {
-			delete(update, *u)
-			continue
+func (w *dnsWatcher) compileUpdate(newAddrs map[string]*Update) []*Update {
+	res := make([]*Update, 0, len(newAddrs)+len(w.curAddrs))
```

Comment:
How about just `var res []*Update`?
Length of updates may be much shorter than `len(newAddrs)+len(w.curAddrs)`.